src/test/run-pass/for-loop-has-unit-body.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    for i in 0..3 {
+        unsafe { std::mem::uninitialized() }
```

Comment:
Not sure this makes sense? I'd rather have a compile-fail test checking there's a type mismatch when there's a body that's not `()`.